org.OpenT2T.Sample.SuperPopular.Thermostat/Nest Thermostat/js/thingTranslator.js

```diff
@@ -0,0 +1,179 @@
+
+// logs device state
+function logDeviceState(device) {
+    if (typeof (device) !== 'undefined') {
+        console.log('  device.name          : ' + device.name);
+        console.log('  device.props         : ' + device.props);
+    } else {
+        console.log('device is undefined');
+    }
+};
+
+var deviceId, accessToken;
+
+// module exports, implementing the schema
+module.exports = {
+
+    device: null,
+
+    initDevice: function(dev) {
+        this.device = dev;
+
+        if (typeof this.device != 'undefined') {
+            if (typeof (this.device.props) !== 'undefined') {
+                var props = JSON.parse(this.device.props);
+
+                if (typeof (props.access_token) !== 'undefined') {
+                    accessToken = props.access_token;
+                } else {
+                    console.log('props.access_token is undefined.');
+                }
+
+                if (typeof (props.id) !== 'undefined') {
+                    deviceId = props.id;
+                } else {
+                    console.log('props.id is undefined.');
+                }
+            } else {
+                console.log('props is undefined.');
+            }
+        } else {
+            console.log('device is undefined.');
+        }
+
+        console.log('javascript initialized.');
+        logDeviceState(this.device);
+    },
+
+    disconnect: function() {
+        console.log('disconnect called.');
+        logDeviceState(this.device);
+    },
+    
+    getTemperatureProperty: function(name, scale, callback) {
+        var options = {
+            protocol: 'https:',
+            host: 'developer-api.nest.com',
+            path: '/devices/thermostats/' + deviceId,
+            headers: {
+                'Authorization': 'Bearer ' + accessToken
+            },
+            method: 'GET'
+        };
+
+        var httpsRequest = https.get(options, function(getRes) {
+            var body = '';
+            getRes.setEncoding('utf8');
+            getRes.on('data', function(data) {
+                body += data;
+            });
+
+            getRes.on('end', function() {
+                if (getRes.statusCode != 200) {
+                    if (callback) {
+                        callback("Error Code:" + getRes.statusCode);
+                        return;
+                    }
+                } else {
+                    var data = JSON.parse(body);
+                    callback(data[name]);
+                    }
+            });
+
+            getRes.on('error', function(e) {
+                callback('error');
+            });
+        });
+
+        httpsRequest.end();
+    },
+
+    setTemperatureProperty: function(name, scale, value, callback) {
+
+        var postData;
+        if (scale == "c"){
+            postData = { name : value };
+        }
+        else {
+            postData = { name : value };
+        }
+        postData = JSON.stringify(postData);
+
+        var options = {
```

Comment:
I shall check it out.

---
In reply to: [68174265](https://github.com/openT2T/translators/pull/22#discussion_r68174265) [](ancestors = 68174265)